core/src/main/java/com/shalan/analytics/core/JsonParameterSerializer.kt

```
@@ -0,0 +1,287 @@
+
+        return try {
+            when (value) {
+                // Collections - try kotlinx serialization first
+                is Collection<*> -> serializeCollection(value)
+                is Map<*, *> -> serializeMap(value)
+                is Array<*> -> serializeArray(value)
+
+                // Try kotlinx serialization for @Serializable classes first
+                else -> serializeWithKotlinx(value) ?: createFallbackJson(value)
+            }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)